docs/migration-guides/v4.md

```diff
@@ -0,0 +1,327 @@
+# Migrating to Tool Kit v4
+
+Version 4 of Tool Kit lets you configure it much more easily for your team's or project's specific use cases, without having to resort to custom plugins as often. Although we've made it as backwards compatible as possible, there are some fundamental changes to its internals and configuration that mean you'll need to update your repo's `.toolkitrc.yml` and any custom plugins when migrating to Tool Kit v4.
+
+> [!NOTE]
+> This migration will take 30m-1hr. If at any point you get stuck or something unexpected happens, [reach out to the Platforms team](https://financialtimes.enterprise.slack.com/archives/C02TRE2V2Q1) and we'll be happy to help.
+
+## 0. Update to latest versions of Tool Kit packages
+
+Update `dotcom-tool-kit` and all the `@dotcom-tool-kit/*` packages you have installed to their latest versions. This can be done with the following command (requires [`jq`](https://jqlang.github.io/jq/)):
+
+```sh
+npm query ":root > [name*=dotcom-tool-kit]" \
+| jq '(.[] | .name + "@latest")' \
+| xargs npm install
```

Comment:
**suggestion**: i've sometimes found that you need to call `npm install --force` instead (boo) might be worth calling that out

> If the `npm install` fails with an error suggesting the use of `--force`, edit the command to call `xargs npm install --force` last instead and run again. This should fix npm's version resolution and shouldn't result in any version errors.

idk something like that

Reply:
so the previous version of the command that's been passed around (the one with `xargs -I{} npm install {}@latest` ran the `npm install`s one by one which is why `--force` was needed. this one should(!) work fine as it does the install in one step. do you think we still need the note?

Reply:
hm i've been using the correct `xargs npm install` version for months now and still need to use `xargs npm install --force` sometimes yeah (verified by checking my terminal history, can see uses from 6 months ago)

Reply:
okay! i'll add the warning